packages/roosterjs-content-model-plugins/lib/imageEdit/ImageEditPlugin.ts

```diff
@@ -0,0 +1,512 @@
+                    ImageEditElementClass.RotateHandle,
+                    Rotator,
+                    () => {
+                        if (
+                            this.imageEditInfo &&
+                            this.selectedImage &&
+                            this.wrapper &&
+                            this.clonedImage
+                        ) {
+                            updateWrapper(
+                                this.imageEditInfo,
+                                this.options,
+                                this.selectedImage,
+                                this.clonedImage,
+                                this.wrapper,
+                                this.rotators
+                            );
+                            this.updateRotateHandleState(
+                                editor,
+                                this.selectedImage,
+                                this.wrapper,
+                                this.rotators,
+                                this.imageEditInfo?.angleRad
+                            );
+                        }
+                    },
+                    this.zoomScale
+                ),
+            ];
+
+            updateWrapper(
+                this.imageEditInfo,
+                this.options,
+                this.selectedImage,
+                this.clonedImage,
+                this.wrapper,
+                this.resizers
+            );
+
+            this.updateRotateHandleState(
+                editor,
+                this.selectedImage,
+                this.wrapper,
+                this.rotators,
+                this.imageEditInfo?.angleRad
+            );
+        }
+    }
+
+    private updateRotateHandleState(
+        editor: IEditor,
+        image: HTMLImageElement,
+        wrapper: HTMLSpanElement,
+        rotators: HTMLDivElement[],
+        angleRad: number | undefined
+    ) {
+        const viewport = editor.getVisibleViewport();
+        const smallImage = isASmallImage(image.width, image.height);
+        if (viewport && rotators && rotators.length > 0) {
+            const rotator = rotators[0];
+            const rotatorHandle = rotator.firstElementChild;
+            if (
+                isNodeOfType(rotatorHandle, 'ELEMENT_NODE') &&
+                isElementOfType(rotatorHandle, 'div')
+            ) {
+                updateRotateHandle(
+                    viewport,
+                    angleRad ?? 0,
+                    wrapper,
+                    rotator,
+                    rotatorHandle,
+                    smallImage
+                );
+            }
+        }
+    }
+
+    public isOperationAllowed(operation: ImageEditOperation): boolean {
+        return operation === 'resize' || operation === 'rotate' || operation === 'flip';
+    }
+
+    public canRegenerateImage(image: HTMLImageElement): boolean {
+        return canRegenerateImage(image) || canRegenerateImage(this.selectedImage);
```

Comment:
This will be very confusing. It checks either the passed in image or current selection. We should make the function behavior predictable, only check one image